hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OMDBCheckpointServlet.java

```diff
@@ -594,9 +594,18 @@ private void writeFilesToArchive(
     if (completed) {
       // Only create the hard link list for the last tarball.
       if (!hardLinkFiles.isEmpty()) {
-        Path hardLinkFile = createHardLinkList(truncateLength, hardLinkFiles);
-        includeFile(hardLinkFile.toFile(), OmSnapshotManager.OM_HARDLINK_FILE,
-            archiveOutputStream);
+        Path hardLinkFile = null;
+        try {
+          hardLinkFile = createHardLinkList(truncateLength, hardLinkFiles);
+          includeFile(hardLinkFile.toFile(), OmSnapshotManager.OM_HARDLINK_FILE,
+              archiveOutputStream);
+        } finally {
+          try {
+            Files.delete(Objects.requireNonNull(hardLinkFile));
```

Comment:
1. It is possible that `hardLinkFile` is null if `createHardLinkList` throws an exception. Logging error in that case is redundant.
2. You can use `FileUtils#deleteQuietly` if exception doesn't matter.

Reply:
Fixed. Added check for `hardLinkFile`. I'd say it is more cleaner to communicate error than just trying to delete a file quietly. Thanks for a suggestion!